api/src/friends/friends.controller.ts

```diff
@@ -0,0 +1,181 @@
+import { Controller, Get, Post, Body, Param, Delete, UseGuards, Request } from '@nestjs/common';
+import { FriendsService } from './friends.service';
+import { CreateFriendDto } from './dto/create-friend.dto';
+import { JwtAuthGuard } from 'src/auth/jwt-auth.guard';
+import { Selectable } from 'kysely';
+import { Friend, FriendRequest } from 'src/types/schema';
+import { ApiBody, ApiCreatedResponse, ApiInternalServerErrorResponse, ApiNotFoundResponse, ApiOkResponse, ApiParam, ApiTags, ApiUnprocessableEntityResponse } from '@nestjs/swagger';
+
+@ApiTags('friends')
+@Controller('friends')
+export class FriendsController {
+  constructor(private readonly friendsService: FriendsService) {}
+
+  //#region <-- Request -->
+  @ApiCreatedResponse({
+    description: "Friend request sent, or added",
+    schema: {
+      type: 'string',
+      examples: {
+        sent: {
+          value: 'Request sent',
+          summary: "Friend request was succesfuly sent"
+        },
+        added: {
+          value: "Friend added",
+          summary: "Request someone who also sent you a request, you are now friend"
+        }
+      }
+    }
+  })
+  @ApiUnprocessableEntityResponse({
+    description: "This status is return if you request as friend yourself or a friend."
+  })
+  @ApiInternalServerErrorResponse({
+    description: "Whenever the backend fail in some point, probably an error with the db."
+  })
+  @ApiBody({
+    type: CreateFriendDto,
+    description: "The user id of who you wish to send a friend request."
+  })
+  @UseGuards(JwtAuthGuard)
+  @Post('request')
+  async addFriend(@Request() req, @Body() createFriendDto: CreateFriendDto): Promise<string> {
+    try {
+      // ? Before creating a friend request, we check if we didn't have a friend request from the target.
+      return await this.friendsService.acceptRequest(createFriendDto.targetId, req.user.id);
+    } catch(NotFoundException) {
+      // ? As we didn't have a friendRequest, we can now create a request.
+      return await this.friendsService.requestAFriend(req.user.id, createFriendDto.targetId);
+    }
+  }
+
+  @ApiOkResponse({
+    description: "An array of every friend request.",
+    schema: {
+      type: 'object',
+      properties: {
+        createdAt: {
+          type: 'string'
+        },
+        friendId: {
+          type: 'integer'
+        },
+        userId: {
+          type: 'integer'
+        }
+      }
+    },
+    isArray: true
+  })
+  @ApiNotFoundResponse({
+    description: "No request found."
+  })
+  @UseGuards(JwtAuthGuard)
+  @Get('request')
+  async findAllRequest(@Request() req): Promise<Selectable<FriendRequest>[]> {
```

Comment:
The front will need more infos about the friends requests (username, avatar, etc)